numpyro/distributions/directional.py

```diff
@@ -0,0 +1,120 @@
+import jax.numpy as np
+from jax import lax, random
+
+from numpyro.distributions import constraints
+from numpyro.distributions.distribution import Distribution
+from numpyro.distributions.util import validate_sample
+from numpyro.util import copy_docs_from
+
+
+@copy_docs_from(Distribution)
+class VonMises(Distribution):
+    arg_constraints = {'location': constraints.real, 'concentration': constraints.positive}
+    support = constraints.real
```

Comment:
I swapped the ordering of adding the location and mapping to [-pi, pi], I'll change that. That should also fix the moving support.

Reply:
Thanks! That would help normalize `log_prob` (integral is 1). Otherwise, this distribution will have improper prior (the density is repeated each 2pi interval).